-t THEME panics, but theme=THEME in config works

Prior to hugo 0.42 I was generating a site using the cli flag `-t THEME`    Using 0.42.+ it panics (see below), but interestingly if I add `theme="THEME"` to my `config.toml` it works.

Regards,

n

```
./bin/hugo version
Hugo Static Site Generator v0.42.1 darwin/amd64 BuildDate: 2018-06-13T10:16:36Z

 ./bin/hugo -t client9
panic: AllThemes not set

goroutine 1 [running]:
github.com/gohugoio/hugo/hugolib/filesystems.createThemesOverlayFs(0xc4200eaa00, 0x0, 0xc42013f0f8, 0x10103e8, 0x30, 0x19ecf20, 0x1, 0xc4201f5920)
	/go/src/github.com/gohugoio/hugo/hugolib/filesystems/basefs.go:572 +0x315
github.com/gohugoio/hugo/hugolib/filesystems.(*sourceFilesystemsBuilder).Build(0xc420065c00, 0xc4201f5920, 0xc4200240c4, 0x3b)
	/go/src/github.com/gohugoio/hugo/hugolib/filesystems/basefs.go:258 +0x2ee
github.com/gohugoio/hugo/hugolib/filesystems.NewBase(0xc4200eaa00, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc42013f390)
	/go/src/github.com/gohugoio/hugo/hugolib/filesystems/basefs.go:232 +0x7e5
github.com/gohugoio/hugo/helpers.NewPathSpecWithBaseBaseFsProvided(0xc420065240, 0x1c1b200, 0xc42019de30, 0x0, 0xc420422fe0, 0x20, 0x20)
	/go/src/github.com/gohugoio/hugo/helpers/pathspec.go:59 +0x9f
github.com/gohugoio/hugo/helpers.NewPathSpec(0xc420065240, 0x1c1b200, 0xc42019de30, 0xc420422fe0, 0x0, 0x210c880)
	/go/src/github.com/gohugoio/hugo/helpers/pathspec.go:41 +0x48
github.com/gohugoio/hugo/deps.New(0xc42042a200, 0xc420065240, 0xc42019de30, 0x1c1b320, 0xc420219860, 0x1c0a3e0, 0x0, 0xc420422fe0, 0x1c0a360, 0xc420422fc0, ...)
	/go/src/github.com/gohugoio/hugo/deps/deps.go:126 +0xb0
github.com/gohugoio/hugo/hugolib.applyDepsIfNeeded(0xc42042a200, 0xc420065240, 0xc42019de30, 0x1c1b320, 0xc420219860, 0x1c0a3e0, 0x0, 0xc420422fe0, 0x1c0a360, 0xc420422fc0, ...)
	/go/src/github.com/gohugoio/hugo/hugolib/hugo_sites.go:190 +0x321
github.com/gohugoio/hugo/hugolib.newHugoSites(0xc42042a200, 0xc420065240, 0x0, 0x1c1b320, 0xc420219860, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/gohugoio/hugo/hugolib/hugo_sites.go:134 +0x24d
github.com/gohugoio/hugo/hugolib.NewHugoSites(0xc42042a200, 0xc420065240, 0x0, 0x1c1b320, 0xc420219860, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/gohugoio/hugo/hugolib/hugo_sites.go:227 +0xf5
github.com/gohugoio/hugo/commands.(*commandeer).loadConfig.func3()
	/go/src/github.com/gohugoio/hugo/commands/commandeer.go:203 +0x117
sync.(*Once).Do(0xc4200967e0, 0xc42013fa98)
	/usr/local/go/src/sync/once.go:44 +0xbe
github.com/gohugoio/hugo/commands.(*commandeer).loadConfig(0xc420096790, 0xc420090001, 0x0, 0x10)
	/go/src/github.com/gohugoio/hugo/commands/commandeer.go:188 +0x562
github.com/gohugoio/hugo/commands.newCommandeer(0x1, 0xc420330060, 0x1c06da0, 0xc4203b5740, 0xc4203b5ac0, 0x0, 0x0, 0x0, 0x1980560, 0xc42001c001, ...)
	/go/src/github.com/gohugoio/hugo/commands/commandeer.go:99 +0x116
github.com/gohugoio/hugo/commands.initializeConfig(0x1980001, 0xc420330060, 0x1c06da0, 0xc4203b5740, 0xc4203b5ac0, 0xc42007fc08, 0xc42007fbd8, 0x0)
	/go/src/github.com/gohugoio/hugo/commands/hugo.go:111 +0x76
github.com/gohugoio/hugo/commands.(*commandsBuilder).newHugoCmd.func1(0xc4203fcc80, 0xc4204931a0, 0x0, 0x2, 0x0, 0x0)
	/go/src/github.com/gohugoio/hugo/commands/commands.go:151 +0x97
github.com/gohugoio/hugo/vendor/github.com/spf13/cobra.(*Command).execute(0xc4203fcc80, 0xc42001c0d0, 0x2, 0x2, 0xc4203fcc80, 0xc42001c0d0)
	/go/src/github.com/gohugoio/hugo/vendor/github.com/spf13/cobra/command.go:756 +0x468
github.com/gohugoio/hugo/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4203fcc80, 0xc4203b5740, 0xc420150460, 0xc42007fee8)
	/go/src/github.com/gohugoio/hugo/vendor/github.com/spf13/cobra/command.go:846 +0x30a
github.com/gohugoio/hugo/commands.Execute(0xc42001c0d0, 0x2, 0x2, 0xc42007ff78, 0x10038ac, 0xc420028118, 0x0)
	/go/src/github.com/gohugoio/hugo/commands/hugo.go:77 +0xbe
main.main()
	/go/src/github.com/gohugoio/hugo/main.go:27 +0x7b
```

--- a/hugolib/config.go
+++ b/hugolib/config.go
@@ -285,6 +285,11 @@ func loadThemeConfig(d ConfigSourceDescriptor, v1 *viper.Viper) ([]string, error
 	if err != nil {
 		return nil, err
 	}
+
+	if len(themeConfigs) == 0 {
+		return nil, nil
+	}
+
 	v1.Set("allThemes", themeConfigs)
 
 	var configFilenames []string

--- a/hugolib/config_test.go
+++ b/hugolib/config_test.go
@@ -42,6 +42,8 @@ func TestLoadConfig(t *testing.T) {
 	assert.Equal("side", cfg.GetString("paginatePath"))
 	// default
 	assert.Equal("layouts", cfg.GetString("layoutDir"))
+	// no themes
+	assert.False(cfg.IsSet("allThemes"))
 }
 
 func TestLoadMultiConfig(t *testing.T) {

